jennmedellin/covalent-quickstart
# Second Code Challenge - Market Dashboard with Material Design

## Description

This code challenge is to demonstrate my creative approach to user
stories, knowledge of Material Design and proficiency using html
components with Angular, as well as GibHub best practices.

#### User Stories

1. As a stock market consumer, I want a dashboard showing the stock
market trends and supporting market news and related stock alerts, so I
can have a quick summary of the market that’s easily consumable

2. As a stock market analyst, I need a wizard-like form with multiple
steps and all the necessary form elements to submit a news story about
the stock market, as if it would add a new market news item to the
dashboard (doesn’t have to be a functional form), so I can add market
news with all the necessary media

#### A/C

a. Use various cards and diverse UI components
b. Follow the Material Design spec for all UI components
c. Use Covalent UI components and any other Angular modules you like
d. Link the two routes together using Angular
e. Use the responsive Flex layout that comes with Covalent
f. Use Git commit best practices for each change/addition
g. Submit a pull request to your own public GitHub repo following PR
best practices

### What's included?

- Dashboard
-- intended for the Stock Market Consumer with market trends, news,
and alerts intended to be easily consumable
- Stories Page
-- intended for Stock Market Analyst
-- contains a mockup view of All Stories and hooks to filter out
drafts from published stories
- Story Form
-- intended to allow Stock Market Analyst user ONLY to generate new
stories and edit existing stories

#### Test Steps / Demo Flow

As a mockup, the code is not fully functional; however, the following
flow should work:
- (if needed) login using any username and password combination
--> "Markets Dashboard" loads with all components
- then open the main navigation at the top left and click "Story Editor"
--> "All Stories" load with static left nav
- then click ... (more) icon at the end of any row
- then click "Edit"
--> "Edit Story" opens with Covalent's stepper component breaking the
form into three steps
- then enter the required data, click on next step title, repeat
- optionally click "publish"
- finally click "Save"
--> "All Stories" load, allowing the analyst to quickly access the
story again, or navigate to the dashboard
